crowbar_framework/app/models/service_object.rb

```diff
@@ -1852,4 +1852,25 @@ def node_changed_roles?(node, old_role, new_role)
     roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
     roles_in_old != roles_in_new
   end
+
+  # return true if no other attribute than the ignored has changed. And that the node has no other
+  # role than the roles in only_for_roles
+  def relevant_attributes_changed_if_roles?(node, old_role, new_role, ignore_attributes, only_for_roles)
+    # sort roles of only_for_roles
+    only_for_roles = only_for_roles.sort
+
+    # return true if node has no other roles that ones in the list (only_for_roles)
+    roles_in_old = old_role.elements.keys.select { |r| old_role.elements[r].include?(node) }.sort
+    roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
+    return true if only_for_roles != roles_in_old || only_for_roles != roles_in_new
+
+    #  return true if the attributes has changed except ignored
+    old_selected_attributes = old_role.default_attributes[@bc_name].reject do |k,v|
+      ignore_attributes.include?(k)
+    end
+    new_selected_attributes = new_role.default_attributes[@bc_name].reject do |k,v|
+      ignore_attributes.include?(k)
+    end
```

Comment:
This works fine until... we need to exclude attributes which are not toplevel. For example, in the provisioner barclamp, we'll want to exclude the `suse.autoyast` attribute (or `role.default_attributes[@bc_name]["suse"]["autoyast"]` if it's clearer).

The way I would deal with it is copy the old and new attributes like this:
```ruby
old_role_attributes = old_role.default_attributes[@bc_name].deep_dup
new_role_attributes = new_role.default_attributes[@bc_name].deep_dup
```

and then remove from these copies the ignored attributes if they exist. The reason I would do it this way is that this makes it much easier to deal with non-toplevel attributes. But I may be wrong :-)

Then we can compare things.